frontend/src/data/variables/BrfssProvider.ts

```diff
@@ -46,39 +48,37 @@ class BrfssProvider extends VariableProvider {
       });
     }
 
-    if (breakdowns.includeTotal && breakdowns.demographic) {
-      const breakdownName: string =
-        breakdowns.demographic === "race"
-          ? "race_and_ethnicity"
-          : breakdowns.demographic;
+    if (
+      breakdowns.demographicBreakdowns.race.enabled &&
+      breakdowns.demographicBreakdowns.race.includeTotal
+    ) {
       const total = df
         .pivot(["state_fips", "state_name"], {
           diabetes_count: (series) => series.sum(),
           diabetes_no: (series) => series.sum(),
           copd_count: (series) => series.sum(),
           copd_no: (series) => series.sum(),
-          [breakdownName]: (series) => "Total",
+          race_and_ethnicity: (series) => "Total",
         })
         .resetIndex();
       df = df.concat(total).resetIndex();
     }
 
-    return new MetricQueryResponse(
-      df
-        .generateSeries({
-          diabetes_per_100k: (row) =>
-            per100k(row.diabetes_count, row.diabetes_count + row.diabetes_no),
-          copd_per_100k: (row) =>
-            per100k(row.copd_count, row.copd_count + row.copd_no),
-        })
-        .toArray()
-    );
+    df = df.generateSeries({
+      diabetes_per_100k: (row) =>
+        per100k(row.diabetes_count, row.diabetes_count + row.diabetes_no),
+      copd_per_100k: (row) =>
+        per100k(row.copd_count, row.copd_count + row.copd_no),
+    });
+
+    return new MetricQueryResponse(df.toArray(), ["brfss"]);
   }
 
   allowsBreakdowns(breakdowns: Breakdowns): boolean {
     const validDemographicBreakdownRequest =
       breakdowns.demographicBreakdownCount() === 0 ||
-      (breakdowns.demographicBreakdownCount() === 1 && breakdowns.race);
+      (breakdowns.demographicBreakdownCount() === 1 &&
+        breakdowns.demographicBreakdowns.race.enabled);
```

Comment:
optional improvement: I feel like many of these checks would be more readable if we had convenience methods like:
```
breakdowns.hasExactlyOneDemographic()
breakdowns.hasOnlyRace()
breakdowns.hasOnlyAge()
...
```